### PR TITLE
Improve Github status checks

### PR DESCRIPTION
**Pull request by kelunik**

### CODE REVIEW

**Comment by kelunik** [comment]
When should `enableForRoom` be called?

**Comment by PeeHaa** [comment]
`enableForRoom` is called on bot init when it's enabled in a room as well as the plugin is manually being enabled in the room.

Now that I think about it we should probably also add the `disableForRoom` method and kill the watcher.

**Comment by DaveRandom** [comment]
@kelunik https://github.com/Room-11/Jeeves/wiki/The-Plugin-API#async-enableforroomchatroom-room-bool-persist

**Comment by kelunik** [comment]
@DaveRandom Persistent storage is TBD be @JayPHP.

File: src/Plugins/Github.php [hunk 1]
```diff
@@ -16,6 +19,7 @@ class Github extends BasePlugin
     private $httpClient;
     private $pluginData;
     private $lastKnownStatus = null;
+    private $rooms = [];
```
Comment (DaveRandom) [inline src/Plugins/Github.php]: `private $enabledRooms` for explicit readability

File: src/Plugins/Github.php [hunk 2]
```diff
@@ -26,12 +30,17 @@ public function __construct(ChatClient $chatClient, HttpClient $httpClient, KeyV
...
-        repeat(
-            function() use ($room) { return $this->checkStatusChange($room); }, 150000
-        );
+        if (!$this->rooms) {
+            repeat(function () {
+                yield $this->checkStatusChange();
+            }, 150000);
+        }
+
+        $this->rooms[] = $room;
```
Comment (DaveRandom) [inline src/Plugins/Github.php]: Here's where there's a change that's actually *required* (the others are just my preference):

`$this->enabledRooms[$room->getIdentifier()->getIdentString()] = $room;`

We will need to handle `disableForRoom()` as well, in which we will remove the target room from the map, and stop the repeater when it's not enabled in any rooms any more.

**Pull request merged**